packages/component-base/test/virtualizer-unlimited-size.test.js

```diff
@@ -202,4 +206,75 @@ describe('unlimited size', () => {
       Math.ceil(itemRect.bottom),
     );
   });
+
+  it('should add more items on size increase', () => {
+    const index = virtualizer.size - 1;
+    virtualizer.scrollToIndex(index);
+    expect(getLastRenderedIndex()).to.equal(index);
+
+    virtualizer.size += 2000;
+    expect(getLastRenderedIndex()).to.be.above(index);
+  });
+
+  it('should remove exceeding items on size decrease', () => {
+    const index = virtualizer.size - 1;
+    virtualizer.scrollToIndex(index);
+    expect(getLastRenderedIndex()).to.equal(index);
+
+    virtualizer.size -= 1000;
+    expect(getLastRenderedIndex()).to.be.below(index);
+  });
+
+  it('should set scroll to end when size decrease affects a visible index', async () => {
+    virtualizer.scrollToIndex(virtualizer.size - 1000);
+    virtualizer.size = virtualizer.firstVisibleIndex - 20;
+    await oneEvent(scrollTarget, 'scroll');
+    const lastItem = elementsContainer.querySelector(`#item-${virtualizer.size - 1}`);
+    expect(lastItem.getBoundingClientRect().bottom).to.be.closeTo(scrollTarget.getBoundingClientRect().bottom, 1);
+  });
+
+  it('should preserve scroll position when size decrease affects a buffered index', () => {
+    // Make sure there are at least 2 buffered items at the end.
+    expect(getLastRenderedIndex() - virtualizer.lastVisibleIndex).to.be.greaterThanOrEqual(2);
+
+    // Scroll to an index and add an additional scroll offset.
+    const index = virtualizer.size - 1000;
+    virtualizer.scrollToIndex(index);
+    scrollTarget.scrollTop += 10;
+
+    // Decrease the size so that the last buffered index exceeds the new size bounds.
+    virtualizer.size = virtualizer.lastVisibleIndex + 1;
+    // await oneEvent(scrollTarget, 'scroll');
+
+    const item = elementsContainer.querySelector(`#item-${index}`);
+    expect(item.getBoundingClientRect().top).to.be.closeTo(scrollTarget.getBoundingClientRect().top - 10, 1);
+  });
+
+  // FIXME: Fails due to a scroll offset reset caused by _adjustVirtualIndexOffset on scroll event.
```

Comment:
Is there a related issue to fix this?

Reply:
Good question. I was planning to create it after the merge because it's only reproducible with the scroll restoration logic.